src/main/java/com/google/research/bleth/servlets/EnqueueExperimentServlet.java

```diff
@@ -0,0 +1,234 @@
+        public static PropertyWrapper create(String property, Number value) {
+            return new AutoValue_EnqueueExperimentServlet_PropertyWrapper(property, value);
+        }
+        public abstract String property();
+        public abstract Number value();
+    }
+
+    private Set<List<PropertyWrapper>> createConfigurations(HttpServletRequest request) {
+        Set<PropertyWrapper> roundsNumValues = createIntValues(request, "roundsNum");
+        Set<PropertyWrapper> rowsNumValues = createIntValues(request, "rowsNum");
+        Set<PropertyWrapper> colsNumValues = createIntValues(request, "colsNum");
+        Set<PropertyWrapper> observersNumValues = createIntValues(request, "observersNum");
+        Set<PropertyWrapper> awakenessCycleValues = createIntValues(request, "awakenessCycle");
+        Set<PropertyWrapper> awakenessDurationValues = createIntValues(request, "awakenessDuration");
+        Set<PropertyWrapper> transmissionThresholdRadiusValues = createDoubleValues(request, "transmissionThresholdRadius");
+
+        Set<List<PropertyWrapper>> configurations = Sets.cartesianProduct(
+                roundsNumValues,
+                rowsNumValues,
+                colsNumValues,
+                observersNumValues,
+                awakenessCycleValues,
+                awakenessDurationValues,
+                transmissionThresholdRadiusValues
+        );
+
+        return configurations.stream().filter(this::validateArguments).collect(ImmutableSet.toImmutableSet());
+    }
+
+    private Set<PropertyWrapper> createIntValues(HttpServletRequest request, String parameter) {
+        int lower = Integer.parseInt(request.getParameter("lower" + upperCaseFirstChar(parameter)));
+        int upper = Integer.parseInt(request.getParameter("upper" + upperCaseFirstChar(parameter)));
+        int step = Integer.parseInt(request.getParameter("step" + upperCaseFirstChar(parameter)));
+        Set<PropertyWrapper> values = new HashSet<>();
+        for (int value = lower; value <= upper; value += step) {
+            values.add(PropertyWrapper.create(parameter, value));
+        }
+        return ImmutableSet.copyOf(values);
+    }
+
+    private Set<PropertyWrapper> createDoubleValues(HttpServletRequest request, String parameter) {
+        double lower = Double.parseDouble(request.getParameter("lower" + upperCaseFirstChar(parameter)));
+        double upper = Double.parseDouble(request.getParameter("upper" + upperCaseFirstChar(parameter)));
+        double step = Double.parseDouble(request.getParameter("step" + upperCaseFirstChar(parameter)));
+        Set<PropertyWrapper> values = new HashSet<>();
+        for (double value = lower; value <= upper; value += step) {
+            values.add(PropertyWrapper.create(parameter, value));
+        }
+        return ImmutableSet.copyOf(values);
+    }
+
+    private String upperCaseFirstChar(String s) {
+        return s.substring(0, 1).toUpperCase() + s.substring(1);
```

Comment:
Note/nit: we recommend to use `Ascii.toUpperCase(s)` rather than `s.toUpperCase()`, as the latter doesn't define a specific language (and may be wrong for some characters, e.g. Turkish "ı" (dotless "i").